### PR TITLE
Replaced flake8 with ruff and added it to CI

### PR DESCRIPTION
**Pull request by GitRon**
Hi @pcraciunoiu 

I realised that flake8 is not run in the CI. Since it's kinda deprecated and [Ruff](https://docs.astral.sh/ruff/) is the new cool kid on the block, I created a PR. Ruff has so many nice add-ons but I didn't want to bloat this PR, so I just used it as a drop-in replacement.

What do you think?

Best from Cologne  
Ronny

### CODE REVIEW

**Comment by pcraciunoiu** [comment]
This looks OK but I'd like to keep the same rules (longer lines mainly). I have to try it locally and see, haven't used ruff yet but I hear good things.

Might take me a couple of weeks to get back to this.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,6 +7,22 @@ on:
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+
+      - name: Install required packages
+        run: pip install ruff
```
Comment (pcraciunoiu) [inline .github/workflows/ci.yml:21]: Should this be in poetry's dev dependencies, and get run on commit? Maybe we can add a pre-commit hook if one doesn't exist already.

**Review by pcraciunoiu — APPROVED** [review]
Finally got to testing this, thanks for the PR!

**Comment by GitRon** [comment]
@pcraciunoiu Awesome, thx! Had totally forgotten about this one 😅

**Pull request merged**